matminer/featurizers/composition/roost.py

```diff
@@ -0,0 +1,569 @@
+
+            writer = SummaryWriter(
+                # NOTE comprhys: I find it useful to automatically label by
+                # time and date but haven't added this here as I imagine that
+                # people won't do as much experimentation with hypers and they
+                # can therefore use model_name to differentiate
+                log_dir=f"runs/{self.model_name}-r{self.run_id}"
+            )
+        else:
+            writer = None
+
+        self.model.fit(
+            train_generator=train_generator,
+            val_generator=None,
+            # val_generator=val_generator,
+            optimizer=self.optimizer,
+            scheduler=self.scheduler,
+            epochs=self.epochs,
+            criterion=self.criterion,
+            normalizer=self.normalizer,
+            model_name=self.model_name,
+            run_id=self.run_id,
+            checkpoint=checkpoint,
+            writer=writer,
+            verbose=verbose,
+        )
+
+        self.fitted = True
+
+        # NOTE move model to cpu as featurisation is faster on the cpu unless
+        # data is batched. Currently featurise_many doesn't use batches and
+        # instead applies the featuriser to each row in turn.
+        self.model.device = "cpu"
+        self.model.to("cpu")
+
+    def featurize(self, *comp):
+
+        if not self.fitted:
+            raise ValueError(f"Please fit the {self} before use")
+        test_set = CompositionData(comp, task=self.task)
+
+        test_generator = DataLoader(test_set, **self.data_params)
+
+        # Ensure model is in evaluation mode
+        self.model.eval()
+
+        with torch.no_grad():
+            for input_, _, _, _ in test_generator:
+                # move tensors to device
+                input_ = (tensor.to(self.model.device) for tensor in input_)
+                # compute intermediate output
+                output = self.model.material_nn(*input_).numpy().ravel()
+                # output = self.model.material_nn(*input_).data.cpu().numpy().ravel()
+
+        return output
+
+    def feature_labels(self):
+        return [f"Roost_feature_{x}" for x in range(self.elem_fea_len)]
+
+    def __str__(self):
+        return "RoostFeaturizer"
+
+    def implementors(self):
+        return ["Rhys Goodall"]
+
+    def citations(self):
+        return [
+            "@article{goodall2019predicting,"
+            "title={Predicting materials properties without crystal structure: "
+            "Deep representation learning from stoichiometry},"
+            "author={Goodall, Rhys EA and Lee, Alpha A},"
+            "journal={arXiv preprint arXiv:1910.00617},"
+            "year={2019}}"
+        ]
+
+
+class CompositionData(Dataset):
+    """
+    The CompositionData dataset is a wrapper for a dataset data points are
+    automatically constructed from composition strings.
+    """
+
+    def __init__(self, comp, targets=None, task="regression", emb="matscholar"):
```

Comment:
Need docstrings for all of these methods, but putting all the Roost utils stuff in one file is a good call.